Revamp and split an internal issue template

darksciencebase opened the pull request:
Splitting the current issue template for internal use into two: one for epics, one for smaller
things that don't need milestones and all that.

More context:
https://metaboat.slack.com/archives/C04CGFHCJDD/p1686548875239219

- Remove unused optional field
- Update title to current convention
- Fix typo
- Remove unused section
- Change capitalization for consistency
- Remove redundant description
- Rename the issue template to mention epic
- Shorten the description of the issue template
- Lighten up the milestones section
- Add an issue template for tasks [ci skip]

darksciencebase committed:
Remove unused optional field

darksciencebase committed:
Update title to current convention

darksciencebase committed:
Fix typo

darksciencebase committed:
Remove unused section

darksciencebase committed:
Change capitalization for consistency

darksciencebase committed:
Remove redundant description

darksciencebase committed:
Rename the issue template to mention epic

darksciencebase committed:
Shorten the description of the issue template

darksciencebase committed:
Lighten up the milestones section

Unfortunately, it doesn't seem possible to prepopulate the issue with an
empty tasklist (https://github.com/orgs/community/discussions/50502).

darksciencebase committed:
Add an issue template for tasks [ci skip]

We had one for epics, with milestones and `.Epic` label. Pretty often
that label was accidentally left on smaller issues that were created
using the same template. Now the smaller issues have their own template
and their own label. Yay.

https://metaboat.slack.com/archives/C04CGFHCJDD/p1686548875239219

darksciencebase committed:
Remove redundant angle brackets

darksciencebase committed:
Use a string to get a valid yaml file

Without the quotes, github says
`Error in user YAML: (<unknown>): did not find expected key while parsing
a block mapping at line 1 column 1`

luizarakaki commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,24 @@
+
+
+***Milestone 1***
+_insert tasklist here_
```
Can we add the actual syntax?

So we don't need to write it down every time :)

```[tasklist]
### Tasks
- [ ] task
```

darksciencebase replied:
yes, but it will not be rendered as the actual tasklist according to https://github.com/orgs/community/discussions/50502. still useful?

luizarakaki replied:
Uh weird... well not very useful then

luizarakaki commented on .github/ISSUE_TEMPLATE/task.md:
```diff
@@ -0,0 +1,14 @@
+
+---
+
+**Links**
```
I'm not sure we will have product docs / eng docs for issues. Most of the time we will link to other epics

darksciencebase replied:
yeah, i think i've only seen a handful. i'll just leave Links as a header then?

darksciencebase replied:
Changed the header to “Context” but left the call-out to related issues in place after all, since I heard from the folks in Success that it's valuable to know when an issue is closed (some customers might be happy to learn that), and if we just create new issues and don't even link the existing related ones, this process breaks

cdeweyx commented on .github/ISSUE_TEMPLATE/epic.md:
```diff
@@ -0,0 +1,24 @@
+
+**Links**
+- product doc: _link to product doc_
+- eng doc: _link to technical design doc_
```
Can probably call this out as optional, not all projects require a technical design doc

albertoperdomo reviewed (approved):
I don't have very specific feedback, because I haven't yet used these. LGTM

darksciencebase committed:
Remove rarely if ever used placeholders

darksciencebase committed:
Call out optional-ness of some links

darksciencebase committed:
Contextualize a header

Context can be some text, a slack thread, a notion page, another issue,
whatever.

darksciencebase committed:
Merge branch 'master' into lena-epic-and-task-issue-templates
